cdk/lib/national-delivery-fulfilment.ts

```diff
@@ -1,22 +1,29 @@
-//import { join } from 'path';
-import { GuScheduledLambda } from '@guardian/cdk';
-import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
-import { GuStack } from '@guardian/cdk/lib/constructs/core';
-import {GuAllowPolicy} from "@guardian/cdk/lib/constructs/iam";
-import type { App } from 'aws-cdk-lib';
-import { Duration } from 'aws-cdk-lib';
-import { Schedule } from 'aws-cdk-lib/aws-events';
-import { Effect, PolicyStatement } from 'aws-cdk-lib/aws-iam';
-import { Runtime } from 'aws-cdk-lib/aws-lambda';
-import { BlockPublicAccess, Bucket } from 'aws-cdk-lib/aws-s3';
-//import { CfnInclude } from 'aws-cdk-lib/cloudformation-include';
+import {GuScheduledLambda} from '@guardian/cdk';
+import type {GuStackProps} from '@guardian/cdk/lib/constructs/core';
+import {GuStack, GuStringParameter} from '@guardian/cdk/lib/constructs/core';
+import {GuAllowPolicy, GuRole} from "@guardian/cdk/lib/constructs/iam";
+import type {App} from 'aws-cdk-lib';
+import {Duration} from 'aws-cdk-lib';
+import {Schedule} from 'aws-cdk-lib/aws-events';
+import {ArnPrincipal, Effect, PolicyStatement} from 'aws-cdk-lib/aws-iam';
+import {Runtime} from 'aws-cdk-lib/aws-lambda';
+import {Bucket} from 'aws-cdk-lib/aws-s3';
 
 export class NationalDeliveryFulfilment extends GuStack {
   constructor(scope: App, id: string, props: GuStackProps) {
     super(scope, id, props);
 
     const app = 'national-delivery-fulfilment';
 
+      const externalRoleArn = new GuStringParameter(
+          this,
+          "ExternalRoleArn",
+          {
+              description:
+                  "Our supplier needs to access our buckets, so we need to tell our stack which role they will use to assume our role.",
+          }
+      );
```

Comment:
It may well be preferable to store this as an SSM param. You can use [this method](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ssm.StringParameter.html#static-valuewbrforwbrstringwbrparameterscope-parametername-version) to retrieve the value at deployment time.

Reply:
done